lib/oban/query.ex

```diff
@@ -84,6 +84,44 @@ defmodule Oban.Query do
     :ok
   end
 
+  @spec retry_all_jobs(Config.t()) :: :ok
+  def retry_all_jobs(conf) do
+    query =
+      Job
+      |> where([j], j.state in ["retryable", "discarded"])
+      |> update([j],
+        set: [
+          state: "available",
+          max_attempts: fragment("GREATEST(?, ? + 1)", j.max_attempts, j.attempt),
+          scheduled_at: ^utc_now(),
+          completed_at: nil,
+          cancelled_at: nil,
+          discarded_at: nil
+        ]
+      )
+
+    Repo.update_all(conf, query, [])
+    :ok
+  end
+
+  @spec retry_all_jobs(Config.t(), Ecto.Queryable.t()) :: :ok
+  def retry_all_jobs(conf, queryable) do
+    query =
+      update(queryable, [j],
```

Comment:
The query should have a `state` condition added to it:

```suggestion
      queryable
      |> where([j], j.state not in ["available", "executing", "scheduled"])
      |> update([j],
```

Interestingly, `retry_job` can use `retry_all_jobs` with a simple `where([j], j.id == ^id)` queryable, too.